main.py

```diff
@@ -30,38 +33,171 @@ def hello_mykhailo():
 def get_time():
     return f"current time: {datetime.datetime.now()}"
 
+
 @app.route("/password")
-def generate_password():
+@use_kwargs(
+    {
+        'length': fields.Int(
+            missing=10,
+            validate=[validate.Range(min=8, max=100)]
+                             )
+    },
+    location='query'
+)
+def generate_password(length):
+
     letters = string.ascii_letters
     digits = string.digits
     special_chars = string.punctuation
 
     conditions = letters + digits + special_chars
-    password_length = random.randint(10, 20)
     password = ''
 
-    for i in range(password_length):
-        password += ''.join(secrets.choice(conditions))
-
-    return password
+    return password.join(random.choices(
+        conditions, k=length)
+                         )
 
 
 @app.route("/parameters")
 def get_average_parameters():
 
     df = pd.read_csv(r'/Users/letsgooo/Desktop/hw.csv')
-    print(df)
-    print(df.columns.tolist())
 
     mean1 = df[' Height(Inches)'].mean()
     mean2 = df[' Weight(Pounds)'].mean()
 
-
     return f'Average height: {mean1}' + '\t' + f'Average weight: {mean2}'
 
 
+@app.route("/students")
+def generate_students():
 
-app.run(port=5001, debug=True)
+    students_count = request.args.get('count', default=100)
+
+    if not students_count.isdigit():
+        return 'ERROR: should be a number'
+
+    students_count = int(students_count)
+
+    if not 1 <= students_count <= 1000:
+        return 'ERROR: should be in range[1, 1000]'
+
+    faker_instance = Faker("UK")
+    student_header = ['first_name', 'last_name', 'email', 'password', 'birthday']
+
+    with open('students.csv', 'w') as file:
+
+        writer = csv.writer(file)
+        writer.writerow(student_header)
+        for _ in range(students_count):
+            student_data = [
+                {
+                    faker_instance.first_name(): 'first_name',
+                    faker_instance.last_name(): 'last_name',
+                    faker_instance.email(): 'email',
+                    faker_instance.password(): 'password',
+                    faker_instance.date(): 'birthday'
+                }
+            ]
+            writer.writerows(student_data)
+
+    students = pd.read_csv('students.csv')
+
+    students.to_html("Table.htm")
```

Comment:
change order:
1. Generate data
2. Save to file
3. Return to the page.

You have redundant step with reading data from file.